run/observer.ts

```diff
@@ -1,28 +1,53 @@
 import type { Observer, Resolve } from "../types.ts";
 
-import { shift } from "../deps.ts";
+import { Computation, shift } from "../deps.ts";
 
 export function createObservable<T>() {
   let observers = new Map<Observer<T>, Resolve<T>>();
 
-  return {
+  let observable = {
     notify(value: T) {
-      for (let listener of observers.values()) {
-        listener(value);
+      let handlers = [...observers.values()];
```

Comment:
We only want the list of handlers that was in place when the event was dispatched. If a handler is installed as part of the handling, it will wait for the next event, so this "slurps" all of the current handlers.

Reply:
Oh my, this is a subtle one.